packages/react-reconciler/src/ReactFiberExpirationTime.new.js

```diff
@@ -18,129 +18,75 @@ import {
-      bucketSizeMs / UNIT_SIZE,
-    )
-  );
-}
-
-// TODO: This corresponds to Scheduler's NormalPriority, not LowPriority. Update
-// the names to reflect.
-export const LOW_PRIORITY_EXPIRATION = 5000;
-export const LOW_PRIORITY_BATCH_SIZE = 250;
-
-export function computeAsyncExpiration(
-  currentTime: ExpirationTime,
-): ExpirationTime {
-  return computeExpirationBucket(
-    currentTime,
-    LOW_PRIORITY_EXPIRATION,
-    LOW_PRIORITY_BATCH_SIZE,
-  );
-}
-
-export function computeSuspenseTimeout(
-  currentTime: ExpirationTime,
-  timeoutMs: number,
-): ExpirationTime {
-  const currentTimeMs = expirationTimeToMs(currentTime);
-  const deadlineMs = currentTimeMs + timeoutMs;
-  return msToExpirationTime(deadlineMs);
-}
-
-// We intentionally set a higher expiration time for interactive updates in
-// dev than in production.
-//
-// If the main thread is being blocked so long that you hit the expiration,
-// it's a problem that could be solved with better scheduling.
-//
-// People will be more likely to notice this and fix it with the long
-// expiration time in development.
-//
-// In production we opt for better UX at the risk of masking scheduling
-// problems, by expiring fast.
-export const HIGH_PRIORITY_EXPIRATION = __DEV__ ? 500 : 150;
-export const HIGH_PRIORITY_BATCH_SIZE = 100;
-
-export function computeInteractiveExpiration(currentTime: ExpirationTime) {
-  return computeExpirationBucket(
-    currentTime,
-    HIGH_PRIORITY_EXPIRATION,
-    HIGH_PRIORITY_BATCH_SIZE,
-  );
-}
+// Accounts for -1 trick to bump updates into a different batch
+const ADJUSTMENT_OFFSET = 5;
 
 export function inferPriorityFromExpirationTime(
-  currentTime: ExpirationTime,
-  expirationTime: ExpirationTime,
+  expirationTime: ExpirationTimeOpaque,
 ): ReactPriorityLevel {
-  if (expirationTime === Sync) {
+  if (expirationTime >= Batched - ADJUSTMENT_OFFSET) {
     return ImmediatePriority;
   }
-  if (expirationTime === Never || expirationTime === Idle) {
-    return IdlePriority;
-  }
-  const msUntil =
-    expirationTimeToMs(expirationTime) - expirationTimeToMs(currentTime);
-  if (msUntil <= 0) {
-    return ImmediatePriority;
-  }
-  if (msUntil <= HIGH_PRIORITY_EXPIRATION + HIGH_PRIORITY_BATCH_SIZE) {
+  if (expirationTime >= UserBlockingUpdateTime - ADJUSTMENT_OFFSET) {
     return UserBlockingPriority;
   }
-  if (msUntil <= LOW_PRIORITY_EXPIRATION + LOW_PRIORITY_BATCH_SIZE) {
+  if (expirationTime >= LongTransition - ADJUSTMENT_OFFSET) {
     return NormalPriority;
   }
 
-  // TODO: Handle LowPriority
+  // TODO: Handle LowPriority. Maybe should give it NormalPriority since Idle is
+  // very agressively deprioritized.
 
   // Assume anything lower has idle priority
   return IdlePriority;
 }
+
+export function isSameOrHigherPriority(
+  a: ExpirationTimeOpaque,
+  b: ExpirationTimeOpaque,
+) {
+  return a >= b;
+}
+
+export function isSameExpirationTime(
+  a: ExpirationTimeOpaque,
+  b: ExpirationTimeOpaque,
+) {
+  return a === b;
+}
```

Comment:
Could I ask that what's the purpose of adding these two helper? Seems not good for code readable, I think maybe we can keep using `>`, `<`, `===`?

Reply:
The purpose of this PR is to prepare for the PR where the data structure changes. (From numbest to a bit mask.)

Reply:
Yea, I think you mean the Lanes, but just like bit masks in `SideEffectTags` or `TypeOfMode`, we don't use any helper, right? Back to the ponit, here I just means for the simple compare like `isSameOrHigher` or `isSame`, we can just use `>=` or `===`, but for complex compare or other purpose like `bumpPriorityHigher `, we can use it as a helper

Reply:
Oh, I know, thansk to the Closure